test/models/request/actions/InboundMessageTransformerSpec.scala

```diff
@@ -54,78 +50,102 @@ import scala.concurrent.Future
 
 class InboundMessageTransformerSpec extends AnyFreeSpec with Matchers with ScalaCheckPropertyChecks with ModelGenerators with OptionValues with ScalaFutures {
 
-  val arrival: Arrival     = arbitrary[Arrival].sample.value
-  val arrivalId: ArrivalId = arrival.arrivalId
+  private val arrival: Arrival     = arbitrary[Arrival].sample.value
+  private val arrivalId: ArrivalId = arrival.arrivalId
 
-  def fakeRequest(code: String, arrivalStatus: ArrivalStatus): ArrivalRequest[AnyContentAsEmpty.type] =
-    ArrivalRequest(FakeRequest("", "").withHeaders("X-Message-Type" -> code), arrival.copy(status = arrivalStatus), web)
-
-  val successfulResponse: Request[AnyContent] => Future[Result] = {
+  private val successfulResponse: Request[AnyContent] => Future[Result] = {
     _ =>
       Future.successful(Ok(HtmlFormat.empty))
   }
 
-  def action = new InboundMessageTransformer()
+  private def action = new InboundMessageTransformer()
 
-  val xmlNegativeAcknowledgement: ArrivalStatus =
+  private val xmlNegativeAcknowledgement: ArrivalStatus =
     Gen.oneOf(Seq(ArrivalXMLSubmissionNegativeAcknowledgement, UnloadingRemarksXMLSubmissionNegativeAcknowledgement)).sample.value
 
-  val responseMessages: Map[String, MessageInbound] = Map(
-    MessageType.ArrivalRejection.code                     -> MessageInbound(ArrivalRejectedResponse, ArrivalRejected),
-    MessageType.UnloadingRemarksRejection.code            -> MessageInbound(UnloadingRemarksRejectedResponse, UnloadingRemarksRejected),
-    MessageType.UnloadingPermission.code                  -> MessageInbound(UnloadingPermissionResponse, UnloadingPermission),
-    MessageType.GoodsReleased.code                        -> MessageInbound(GoodsReleasedResponse, GoodsReleased),
-    MessageType.XMLSubmissionNegativeAcknowledgement.code -> MessageInbound(XMLSubmissionNegativeAcknowledgementResponse, xmlNegativeAcknowledgement)
-  )
-
   "InboundMessageTransformer" - {
 
-    "handle unsupported `X-Message-Type`" in {
+    "returns 200 for supported `X-Message-Type`" in {
+      forAll(arbitrary[ChannelType]) {
+        channel =>
+          val arrivalMovement = arrival.copy(status = ArrivalStatus.ArrivalSubmitted)
+
+          val request =
+            ArrivalRequest(FakeRequest("", "").withHeaders("X-Message-Type" -> MessageType.ArrivalRejection.code), arrivalMovement, channel)
```

Comment:
Do we maybe want to generate the MessageType to make this test a bit more robust (or is that overkill?)

Reply:
I don't think it is possible :( since there is validation on the X-Message-Type being a known value AND it must be a value that we know how to handle given all the messages we've see so far. Any generators to make this possible are likely to push the code into the unintelligible region